src/main/java/org/wildfly/security/auth/provider/oauth2/OAuth2Util.java

```diff
@@ -0,0 +1,194 @@
+
+        return null;
+    }
+
+    /**
+     * Returns a {@link Attributes} instance based on the given {@link JsonObject}.
+     *
+     * @param claims a json object with the claims to extract
+     * @return an {@link Attributes} instance with attributes from the given json object
+     */
+    static Attributes toAttributes(JsonObject claims) {
+        return claims.entrySet().stream().reduce(new MapAttributes(), (mapAttributes, entry) -> {
+            String claimName = entry.getKey();
+            JsonValue claimValue = entry.getValue();
+
+            if (JsonValue.ValueType.ARRAY.equals(claimValue.getValueType())) {
+                JsonArray jsonArray = claims.getJsonArray(claimName);
+                jsonArray.forEach(arrayValue -> mapAttributes.addLast(claimName, asString(arrayValue)));
+            } else {
+                mapAttributes.addLast(claimName, asString(claimValue));
+            }
+
+            return mapAttributes;
+        }, (mapAttributes, mapAttributes2) -> mapAttributes);
+    }
+
+    private static String asString(JsonValue value) {
+        if (JsonValue.ValueType.STRING.equals(value.getValueType())) {
+            return ((JsonString) value).getString();
+        }
+
+        return value.toString();
+    }
+
+    private static HttpURLConnection openConnection(URL url) {
+        Assert.assertNotNull(url);
+        HttpURLConnection connection = null;
+
+        try {
+            connection = (HttpURLConnection) url.openConnection();
+
+            if (url.getProtocol().equalsIgnoreCase("https")) {
+                try {
+                    HttpsURLConnection https = (HttpsURLConnection) connection;
+                    SSLContext sslContext = SSLContext.getInstance("TLS");
+
+                    sslContext.init(null, new TrustManager[] {
+                            new X509TrustManager()                            {
+                                public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                                    return null;
+                                }
+                                public void checkClientTrusted( java.security.cert.X509Certificate[] certs, String authType) {
+
+                                }
+                                public void checkServerTrusted( java.security.cert.X509Certificate[] certs, String authType) {
+
+                                }
+                            }
+                    }, null);
+
+                    https.setSSLSocketFactory(sslContext.getSocketFactory());
+                    https.setHostnameVerifier((hostname, session) -> true);
+                } catch( Exception e) {
+                    throw new RuntimeException("Error while preparing HTTPS connection for token introspection.", e);
+                }
+            }
+
+            return connection;
+        } catch (IOException ioe) {
+            throw new RuntimeException("Could not create http connection to URL [" + url + "]");
```

Comment:
These kinds of exceptions can leak to logs so really should be in ElytronMessages - however for this method throwing an IOException seems quite appropriate if it can't proceed.